hermes-client/src/main/java/pl/allegro/tech/hermes/client/restTemplate/RestTemplateHermesSender.java

```diff
@@ -1,86 +0,0 @@
-package pl.allegro.tech.hermes.client.restTemplate; // CHECKSTYLE.SUPPRESS: PackageName
-
-import org.springframework.http.HttpEntity;
-import org.springframework.http.ResponseEntity;
-import org.springframework.util.LinkedMultiValueMap;
-import org.springframework.util.concurrent.ListenableFutureCallback;
-import org.springframework.web.client.AsyncRestTemplate;
-import org.springframework.web.client.HttpStatusCodeException;
-import pl.allegro.tech.hermes.client.HermesMessage;
-import pl.allegro.tech.hermes.client.HermesResponse;
-import pl.allegro.tech.hermes.client.HermesSender;
-
-import java.net.URI;
-import java.util.Map;
-import java.util.TreeMap;
-import java.util.concurrent.CompletableFuture;
-
-import static java.util.stream.Collectors.toMap;
-import static pl.allegro.tech.hermes.client.HermesResponseBuilder.hermesResponse;
-
-/**
- * RestTemplateHermesSender.
- *
- * @deprecated as of Hermes 2.2.10, in favor of {@link pl.allegro.tech.hermes.client.webclient.WebClientHermesSender}
```

Comment:
> is it impossible to keep support for RestTemplateHermesSender with newer Spring?
Yes. `RestTemplateHermesSender` should have actually been called `AsyncRestTemplateHermesSender` because it was using `AsyncRestTemplateHermesSender` underneath. `AsyncRestTemplate` was removed in spring boot 3. In theory we could switch to using regular `RestTemplate` underneath but `HermesSender` is inherently asynchronous:

```java
public interface HermesSender {

    CompletableFuture<HermesResponse> send(URI uri, HermesMessage message);
}
```

and `RestTemplate` is sync. Imo its best to remove `RestTemplateHermesSender` and encoure clients to migrate to `WebClientSender`